linker/Mono.Linker/Dependencies.cs

```diff
@@ -0,0 +1,274 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Cecil;
+using Mono.Linker.Steps;
+
+namespace Mono.Linker
+{
+	public class Dependencies
+	{
+		protected readonly LinkContext _context;
+		Stack<object> dependency_stack;
+		System.Xml.XmlWriter writer;
+		GZipStream zipStream;
+
+		public Dependencies(LinkContext context)
+		{
+			_context = context;
+		}
+
+		public void PrepareDependenciesDump()
+		{
+			PrepareDependenciesDump("linker-dependencies.xml.gz");
+		}
+
+		public void PrepareDependenciesDump(string filename)
+		{
+			dependency_stack = new Stack<object>();
+			System.Xml.XmlWriterSettings settings = new System.Xml.XmlWriterSettings();
+			settings.Indent = true;
+			settings.IndentChars = "\t";
+			var depsFile = File.OpenWrite(filename);
+			zipStream = new GZipStream(depsFile, CompressionMode.Compress);
+
+			writer = System.Xml.XmlWriter.Create(zipStream, settings);
+			writer.WriteStartDocument();
+			writer.WriteStartElement("dependencies");
+			writer.WriteStartAttribute("version");
+			writer.WriteString("1.0");
+			writer.WriteEndAttribute();
+		}
+
+		public IDisposable BeginScope(TypeDefinition definition)
```

Comment:
I was thinking of renaming these to simply "Begin".  I didn't want to use the term "Push" because the "Push" method that exists today will need to remain and I didn't we'd want to have different "Push" methods that behave differently.  Although, I guess in a way if you don't dispose of the result of "Begin" then it does behave the same as "Push".  So I'm open to calling them "Push" also if that is preferred